modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/Swift5ClientCodegen.java

```diff
@@ -214,14 +214,14 @@ public Swift5ClientCodegen() {
         typeMapping.put("float", "Float");
         typeMapping.put("number", "Double");
         typeMapping.put("double", "Double");
-        typeMapping.put("file", "URL");
-        typeMapping.put("binary", "URL");
+        typeMapping.put("file", "Data");
+        typeMapping.put("binary", "Data");
```

Comment:
I think this is more of a bug fix than a breaking change

Reply:
This is the same discussion as https://github.com/OpenAPITools/openapi-generator/pull/8515#discussion_r609716185

Reply:
Now only `binary` maps to `Data`

Reply:
@aymanbagabas could you please move this change to another PR, so that we can unlock this PR? Thanks

Reply:
@aymanbagabas I'd appreciate if you can create a new PR for this change. @4brunu he is right that binary needs to be data and probably file too.

You can't parse a binary response into a URL like it's done currently. The binary content is given as URL path to the URL constructor which doesn't make sense at all.

Reply:
After some investigation, the test issue when changing the type binary from `URL` to `Data`, is related to the code in ModelUtils line 659.

https://github.com/OpenAPITools/openapi-generator/blob/3894aa47591f71406bed4c8da70bc01b6823023e/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java#L2059-L2062

https://github.com/OpenAPITools/openapi-generator/blob/3894aa47591f71406bed4c8da70bc01b6823023e/modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java#L643-L660

The binary type will be considered as file type.

To add some interesting bits to the discussion
The Kotlin Client generator, considers binary as BiteArray (or Data in Swift) and file as java.io.File (or URL in Swift)

https://github.com/OpenAPITools/openapi-generator/blob/3894aa47591f71406bed4c8da70bc01b6823023e/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractKotlinCodegen.java#L159-L165

The Java Client generator, considers binary as byte[], and file as java.io.File.

Since this is a big change on how files/binary is treated in the Swift Generator that is also a breaking change, I just would like to be sure that this is the right move, but let's continue this discussion on the new PR.